### ISSUE TITLE
Handling Connections and pagination

### ISSUE THREAD

**Issue by MohGanji** [Urigo/SOFA#50]
I replied to the medium post about SOFA and @kamilkisiela asked me to open an issue here with an example.

Right now graphql uses connections for pagination, i.e. a type with arguments like`before`, `after`, `first` and `last` to determine what page you want to fetch from a long list of items.

[graphql-relay](https://github.com/graphql/graphql-relay-js) provides these connections in a simple way that you only need to define a connection for a type, then use the connection, on the client side, relay can use this predefined connection type to fetch pages.

connection types are like this in graphql queries:
```
query {
  chapters(
      after:"YXJyYXljb25uZWN0aW9uOjQ=", // --> some curser
      first:3 // --> fetch first three after given curser
){
    edges{
      node{
        id
        name
      }
    }
    pageInfo{ // --> details about this page
      hasNextPage
      hasPreviousPage
      startCursor
      endCursor
    }
  }
}
```

so this is how graphql currently supports pagination and it is similar in rest, it was a concern for me if I wanted to use sofa in a project, but I didn't find anything related to pagination in SOFA.

**Comment by wtrocki** [comment]
I'm very biased here but I think GraphQL.js/Apollo is missing Pagination abstraction for GraphQL - if that will be there then Sofa might simply adapt it properly for the REST standard. As long as Sofa works with some of the pagination standards like one in https://www.tortilla.academy/Urigo/WhatsApp-Clone-Tutorial/master/next/step/0 it

**Comment by Urigo** [comment]
@MohGanji I'm not sure what is the issue here...
If you create a GraphQL API, add grpahql-relay to it and generate REST api using SOFA, what are you missing?
can you create a reproduction?
Thanks

**Comment by ardatan** [comment]
Feel free to open a new issue if you have further questions.
